domain/modelagent/state/state.go

```diff
@@ -75,3 +76,55 @@ WHERE uuid = $M.model_id
 	}
 	return vers, nil
 }
+
+// AgentVersionForModelName returns the agent version for the model with the
+// given name and owner. If such a model cannot be found, an error satisfying
+// [modelerrors.NotFound] will be returned.
+func (st *State) AgentVersionForModelName(
```

Comment:
Why not have a common method for running a provided prepared query and getting the result/version which is used by both AgentVersionForModelName and GetModelAgentVersion?

Reply:
the query is different. One has
```
WHERE uuid = $modelIDArgs.model_id
```
and one has
```
WHERE name = $modelNameArgs.model_name
AND owner_name = $modelNameArgs.owner
```